server/server.go

```diff
@@ -1213,6 +1214,7 @@ func (s *Server) campaignLeader() {
 	ctx, cancel := context.WithCancel(s.serverLoopCtx)
 	defer cancel()
 	defer s.member.ResetLeader()
```

Comment:
These two lines should be removed now.

Reply:
I think we still need to cancel the context when there is a problem before we reach the below defer function.

Reply:
yes, some branches will return early.

Reply:
But it will be reset and canceled twice.

Reply:
yes, it's valid.